browser-test/src/application_navigation.test.ts

```diff
@@ -1438,5 +1438,161 @@ describe('Applicant navigation flow', () => {
+        questionName: questionAddress,
+        questionText: questionAddress,
+      })
+
+      await adminQuestions.addTextQuestion({
+        questionName: questionText1,
+        questionText: questionText1,
+      })
+
+      await adminQuestions.addTextQuestion({
+        questionName: questionText2,
+        questionText: questionText2,
+      })
+
+      // Create Program
+      await adminPrograms.addProgram(programName)
+
+      // Attach questions to program
+      await adminPrograms.editProgramBlock(programName, screen1, [
+        questionAddress,
+      ])
+
+      await adminPrograms.addProgramBlock(programName, screen2, [questionText1])
+
+      await adminPrograms.addProgramBlock(programName, screen3, [questionText2])
+
+      await adminPrograms.goToBlockInProgram(programName, screen1)
+
+      await adminPrograms.clickAddressCorrectionToggleByName(questionAddress)
+
+      const addressCorrectionInput =
+        adminPrograms.getAddressCorrectionToggleByName(questionAddress)
+
+      // TODO: Change this to true when done with other PR 6598 before merging this PR
+      expect(await addressCorrectionInput.inputValue()).toBe('false')
+
+      // Set thing to soft eligibilty
+      await adminPrograms.toggleEligibilityGating()
+
+      // Add address eligibility predicate
+      await adminPrograms.goToEditBlockEligibilityPredicatePage(
+        programName,
+        screen1,
+      )
+
+      await adminPredicates.addPredicates([
+        {
+          questionName: questionAddress,
+          scalar: 'service_area',
+          operator: 'in service area',
+          values: ['Seattle'],
+        },
+      ])
+
+      // Add the address visibility predicate
+      await adminPrograms.goToBlockInProgram(programName, screen2)
+
+      await adminPrograms.goToEditBlockVisibilityPredicatePage(
+        programName,
+        screen2,
+      )
+
+      await adminPredicates.addPredicates([
+        {
+          questionName: questionAddress,
+          action: 'shown if',
+          scalar: 'service_area',
+          operator: 'in service area',
+          values: ['Seattle'],
+        },
+      ])
+
+      // Publish Program
+      await adminPrograms.gotoAdminProgramsPage()
+      await adminPrograms.publishProgram(programName)
+
+      await logout(page)
+    })
+
+    it('when address is eligible show hidden screen', async () => {
+      const {page, applicantQuestions} = ctx
+      await enableFeatureFlag(page, 'esri_address_correction_enabled')
+      await applicantQuestions.applyProgram(programName)
+
+      // Fill out application and submit.
+      await applicantQuestions.answerAddressQuestion(
+        'Legit Address',
+        '',
+        'Seattle',
+        'WA',
+        '98109',
+        0,
+      )
+      await applicantQuestions.clickNext()
+      await applicantQuestions.expectVerifyAddressPage(true)
+      await applicantQuestions.clickNext()
+      await applicantQuestions.answerTextQuestion('answer 1')
```

Comment:
suggestion (non-blocking): Add an expect here that tests that the screen showing is indeed the hidden screen. Alternatively, add a comment saying something like "this screen will be hidden if the address is not eligible".